fhir-persistence-jdbc/src/main/java/org/linuxforhealth/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -1002,6 +1003,59 @@ public MultiResourceResult search(FHIRPersistenceContext context, Class<? extend
         }
     }
 
+    /**
+     * Remove the additional resources fetched at the beginning and end of the search results for validating the expected first and last search results. 
+     * Validate the input expected first resource Id and last resource Id with the search results.
+     * If the expected first resource Id or last resource Id do not match with the search results then add a OperationOutcome with Issue Severity Warning to the result Builder.
+     * @param resourceDTOList the list of 'match' resources
+     * @param searchContext the current search context
+     * @param resultBuilder The MultiResourceResult builder
+     * @return the list of filtered resources after removing the additional resources.
+     */
+    private List<org.linuxforhealth.fhir.persistence.jdbc.dto.Resource>
+        validateExpectedSearchPagingResults(List<org.linuxforhealth.fhir.persistence.jdbc.dto.Resource> resourceDTOList, FHIRSearchContext searchContext, Builder resultBuilder) {
+        org.linuxforhealth.fhir.persistence.jdbc.dto.Resource firstResourceResult = null;
+        org.linuxforhealth.fhir.persistence.jdbc.dto.Resource lastResourceResult = null;
+        if(resourceDTOList != null) {
+            if (resourceDTOList.size() > 0 && searchContext.getPageNumber() != 1){
+                firstResourceResult = resourceDTOList.get(0);
+                resourceDTOList.remove(0);
+                resultBuilder.expectedPreviousId(firstResourceResult.getLogicalId());
+                
+            } 
+            if (resourceDTOList.size() > 0 && (resourceDTOList.size() > searchContext.getPageSize())) {
+                lastResourceResult = resourceDTOList.get(resourceDTOList.size() - 1);
+                resourceDTOList.remove(resourceDTOList.size() - 1);
+                resultBuilder.expectedNextId(lastResourceResult.getLogicalId());
+            }
+            
+            if (firstResourceResult != null && searchContext.getLastId() != null && !searchContext.getLastId().equals(firstResourceResult.getLogicalId())) {
```

Comment:
I think in my example, the values being checked are supposed to be the values from the current page. So if the search context specifies a non-null firstId value, then the value should match the id of the first resource belonging to this page (which is resourceDTOList.get(0) AFTER the first item was removed.

Similarly, if lastId is non-null in the search context, then its value should match the id of the last resource belonging to this page (which is resourceDTOList.get(resourceDTOList.size()-1) AFTER the last item was removed.